collector: Problematic use of process.ClockTicks

The code:
https://github.com/pingcap/tidb-insight/blob/227008e9835bd6f72fd5396073cdc6f738b5afbc/collector/insight/process.go#L168

The issue with this is that this was never meant to be exposed and has been removed in v3 of gopsutil. See https://github.com/shirou/gopsutil/issues/1197 for details.

This causes:
- Build failures on FreeBSD
- Issues with upgrading to gopsutil v3

I was able to compile with this patch (not checked for correctness):
```diff
diff --git a/collector/pd.go b/collector/pd.go
index 0529dc9..23b6e28 100644
--- a/collector/pd.go
+++ b/collector/pd.go
@@ -8,7 +8,7 @@ import (
        "strconv"
        "strings"
 
-       "github.com/shirou/gopsutil/process"
+       "github.com/shirou/gopsutil/v3/process"
 )
 
 // PDMeta is the metadata struct of a PD server
diff --git a/collector/process.go b/collector/process.go
index 7e9165b..7c6d8a0 100644
--- a/collector/process.go
+++ b/collector/process.go
@@ -7,8 +7,8 @@ import (
        "strconv"
        "strings"
 
-       "github.com/shirou/gopsutil/cpu"
-       "github.com/shirou/gopsutil/process"
+       "github.com/shirou/gopsutil/v3/cpu"
+       "github.com/shirou/gopsutil/v3/process"
 )
 
 // ProcessStat contains basic info of a running process
@@ -165,7 +165,7 @@ func getProcStartTime(proc *process.Process) (float64, error) {
        }
        fields := strings.Fields(string(contents))
        if startTime, err := strconv.ParseFloat(fields[21], 64); err == nil {
-               return startTime / process.ClockTicks, err
+               return startTime / 100, err
        }
        return 0, err
 }
@@ -175,7 +175,8 @@ func (proc_stat *ProcessStat) getProcessStat(proc *process.Process) {
        proc_stat.Name, _ = proc.Name()
        proc_stat.Exec, _ = proc.Exe()
        proc_stat.Cmdline, _ = proc.Cmdline()
-       proc_stat.Status, _ = proc.Status()
+       s, _ := proc.Status()
+       proc_stat.Status = s[0]
        proc_stat.StartTime, _ = getProcStartTime(proc)
        proc_stat.CPUTimes, _ = proc.Times()
        proc_stat.Memory, _ = proc.MemoryInfo()
diff --git a/collector/tidb.go b/collector/tidb.go
index 85d0698..5c6c8b5 100644
--- a/collector/tidb.go
+++ b/collector/tidb.go
@@ -8,7 +8,7 @@ import (
        "strconv"
        "strings"
 
-       "github.com/shirou/gopsutil/process"
+       "github.com/shirou/gopsutil/v3/process"
 )
 
 // TiDBMeta is the metadata struct of TiDB server
diff --git a/collector/tikv.go b/collector/tikv.go
index 32dfce2..c9ac1b8 100644
--- a/collector/tikv.go
+++ b/collector/tikv.go
@@ -8,7 +8,7 @@ import (
        "strconv"
        "strings"
 
-       "github.com/shirou/gopsutil/process"
+       "github.com/shirou/gopsutil/v3/process"
 )
 
 // TiKVMeta is the metada struct of a TiKV server
```

I can compile this on FreeBSD without switching to v3 with this patch:
```diff
diff --git a/collector/process.go b/collector/process.go
index 7e9165b..6ceff3c 100644
--- a/collector/process.go
+++ b/collector/process.go
@@ -165,7 +165,7 @@ func getProcStartTime(proc *process.Process) (float64, error) {
        }
        fields := strings.Fields(string(contents))
        if startTime, err := strconv.ParseFloat(fields[21], 64); err == nil {
-               return startTime / process.ClockTicks, err
+               return startTime / 100, err
        }
        return 0, err
 }
```

I also noticed that:

- There is no `go.mod` and `go.sum`
- While building this warning is shown: `go: -i flag is deprecated`

I found this while trying to compile "tiup cluster"
```
% gmake cluster
GO111MODULE=on CGO_ENABLED=0 GOOS=freebsd GOARCH=amd64 go build  -ldflags '-w -s -X "github.com/pingcap/tiup/pkg/version.GitHash=e8fe1107" -X "github.com/pingcap/tiup/pkg/version.GitRef=master"' -o bin/tiup-cluster ./components/cluster
# github.com/pingcap/tidb-insight/collector/insight
../../../go/pkg/mod/github.com/pingcap/tidb-insight/collector@v0.0.0-20220111101533-227008e9835b/insight/process.go:168:38: undefined: process.ClockTicks
gmake: *** [Makefile:66: cluster] Error 2
```

--- a/collector/insight/process.go
+++ b/collector/insight/process.go
@@ -2,7 +2,6 @@
 package insight
 
 import (
-	"io/ioutil"
 	"log"
 	"strconv"
 	"strings"
@@ -154,20 +153,6 @@ func getRlimitUsage(proc *process.Process) []RlimitUsage {
 		result = append(result, usage)
 	}
 	return result
-}
-
-func getProcStartTime(proc *process.Process) (float64, error) {
-	statPath := GetProcPath(strconv.Itoa(int(proc.Pid)), "stat")
-	contents, err := ioutil.ReadFile(statPath)
-	if err != nil {
-		log.Fatal(err)
-		return 0, err
-	}
-	fields := strings.Fields(string(contents))
-	if startTime, err := strconv.ParseFloat(fields[21], 64); err == nil {
-		return startTime / float64(process.ClockTicks), err
-	}
-	return 0, err
 }
 
 func (proc_stat *ProcessStat) getProcessStat(proc *process.Process) {

--- a/collector/insight/process_linux.go
+++ b/collector/insight/process_linux.go
@@ -1,0 +1,23 @@
+//go:build linux
+// +build linux
+
+package insight
+
+import (
+	"github.com/shirou/gopsutil/process"
+	"io/ioutil"
+)
+
+func getProcStartTime(proc *process.Process) (float64, error) {
+	statPath := GetProcPath(strconv.Itoa(int(proc.Pid)), "stat")
+	contents, err := ioutil.ReadFile(statPath)
+	if err != nil {
+		log.Fatal(err)
+		return 0, err
+	}
+	fields := strings.Fields(string(contents))
+	if startTime, err := strconv.ParseFloat(fields[21], 64); err == nil {
+		return startTime / float64(process.ClockTicks), err
+	}
+	return 0, err
+}

--- a/collector/insight/process_linux.go
+++ b/collector/insight/process_linux.go
@@ -4,8 +4,12 @@
 package insight
 
 import (
-	"github.com/shirou/gopsutil/process"
 	"io/ioutil"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/shirou/gopsutil/process"
 )
 
 func getProcStartTime(proc *process.Process) (float64, error) {

--- a/collector/insight/process_others.go
+++ b/collector/insight/process_others.go
@@ -1,0 +1,12 @@
+//go:build !linux
+// +build !linux
+
+package insight
+
+import (
+	"github.com/shirou/gopsutil/process"
+)
+
+func getProcStartTime(proc *process.Process) (float64, error) {
+	return 0, nil
+}

